js/views/components/CryptoCurSelector.js

```diff
@@ -0,0 +1,159 @@
+    }
+
+    if (!options.initialState.currencies ||
+      !Array.isArray(options.initialState.currencies)) {
+      throw new Error('Please provide an initial array of currencies.');
+    }
+
+    if (options.initialState.activeCurs &&
+      !Array.isArray(options.initialState.activeCurs)) {
+      throw new Error('If activeCurs are provided, they must be an array.');
+    }
+
+    if (options.initialState.disabledCurs &&
+      !Array.isArray(options.initialState.disabledCurs)) {
+      throw new Error('If disabledCurs are provided, they must be an array.');
+    }
+
+    const disabledCurs = options.initialState.disabledCurs ?
+      options.initialState.disabledCurs :
+      options.initialState.currencies.filter(c => !isSupportedWalletCur(c));
+
+    const opts = {
+      ...options,
+      initialState: {
+        controlType: 'checkbox',
+        currencies: [],
+        activeCurs: [],
+        disabledCurs,
+        sort: false,
+        ...options.initialState,
+      },
+    };
+
+    super(opts);
+  }
+
+  get className() {
+    return 'unstyled borderStackedAll curSelector';
+  }
+
+  get tagName() {
+    return 'ul';
+  }
+
+  events() {
+    return {
+      'click .js-curControl': 'handleCurClick',
+    };
+  }
+
+  handleCurClick(e) {
+    const code = $(e.target).attr('data-code');
+    let activeCurs = [...this.getState().activeCurs];
+    // Toggle the current active state when clicked.
+    const nowActive = !activeCurs.includes(code);
+
+    if (this.getState().controlType === 'radio') {
+      activeCurs = [code];
+    } else {
+      if (nowActive) activeCurs.push(code);
+      else activeCurs = activeCurs.filter(c => c !== code);
+    }
+
+    this.trigger('currencyClicked', {
+      currency: code,
+      active: nowActive,
+      activeCurs,
+    });
+
+    this.setState({
+      activeCurs,
+    });
+  }
+
+  setState(state = {}, options = {}) {
+    const curState = this.getState();
+    // De-dupe any passed in currencies.
+    state.currencies = state.currencies && Array.isArray(state.currencies) ?
+      [...new Set(state.currencies)] : [];
+
+    state.activeCurs = state.activeCurs && Array.isArray(state.activeCurs) ?
+      [...new Set(state.activeCurs)] : [];
+
+    // Remove any disabled currencies from the active list.
+    if (state.disabledCurs && state.disabledCurs.length) {
+      state.activeCurs = state.activeCurs.filter(c =>
+        !state.disabledCurs.includes(c));
+    }
+
+    // Radio controls can only have one active currency.
+    if (state.activeCurs.length && state.controlType === 'radio') {
+      state.activeCurs = [state.activeCurs[0]];
+    }
+
+    const processedState = {
+      ...state,
+      // This is a derived field and should not be directly set
+      processedCurs: Array.isArray(curState.processedCurs) ?
+        curState.processedCurs : [],
+    };
+
+    // If news currencies have been passed in, replace the old ones.
+    if (processedState.currencies.length &&
```

Comment:
Do you mean state instead of curState?

curState is the state before the state was changed, state is the new passed in state.

The isArray isn't needed, because state.currencies is checked for that on line 93, it will always be an array after that line.

Reply:
yes, I meant `state`. If they're not even passing in `state.currencies`, let's avoid the `isEqual` call.